### PR TITLE
[WIP] Fastlog

### PR DESCRIPTION
**Pull request by fgregg**
hi @dirko, just wanted to show you some speed up from using fast approximations of `log` and `exp` As I mentioned in #2, `logaddexp` was the largest, single remaining bottleneck. It's a potential speed up of 50%

With the numpy's `logaddexp`

``` bash
         3922754 function calls (3922633 primitive calls) in 28.223 seconds

   Ordered by: cumulative time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.142    0.142   28.223   28.223 speed.py:1(<module>)
   100000    0.316    0.000   27.991    0.000 __init__.py:62(__call__)
   100000    0.803    0.000   22.043    0.000 pyhacrf.py:140(predict_proba)
   100000    1.230    0.000   17.298    0.000 pyhacrf.py:272(predict)
   100000   13.567    0.000   13.567    0.000 {pyhacrf.algorithms.forward_predict}
```

with a cython `logaddexp`, and `fastlog` and `fastexp`

``` bash
         3922754 function calls (3922633 primitive calls) in 22.543 seconds

   Ordered by: cumulative time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.136    0.136   22.543   22.543 speed.py:1(<module>)
   100000    0.314    0.000   22.329    0.000 __init__.py:62(__call__)
   100000    0.802    0.000   16.538    0.000 pyhacrf.py:140(predict_proba)
   100000    1.217    0.000   11.800    0.000 pyhacrf.py:272(predict)
   100000    8.160    0.000    8.160    0.000 {pyhacrf.algorithms.forward_predict}
```

with a cython `logaddexp`, and `fasterlog` and `fasterexp`

``` bash
         3922754 function calls (3922633 primitive calls) in 19.757 seconds

   Ordered by: cumulative time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.144    0.144   19.757   19.757 speed.py:1(<module>)
   100000    0.318    0.000   19.547    0.000 __init__.py:62(__call__)
   100000    0.814    0.000   13.567    0.000 pyhacrf.py:140(predict_proba)
   100000    1.248    0.000    8.733    0.000 pyhacrf.py:272(predict)
   100000    0.061    0.000    5.639    0.000 feature_extraction.py:96(fit_transform)
   100000    0.163    0.000    5.577    0.000 feature_extraction.py:112(transform)
   100000    1.442    0.000    5.414    0.000 feature_extraction.py:128(_extract_features)
   100000    4.990    0.000    4.990    0.000 {pyhacrf.algorithms.forward_predict}
```

BTW, I evaluated cephes and it didn't really help.

### CODE REVIEW

File: pyhacrf/algorithms.pyx [hunk 1]
```diff
@@ -16,29 +18,34 @@ cpdef dict forward(np.ndarray[np.int64_t, ndim=2] lattice, np.ndarray[np.float64
...
-        edge_potential = (x_dot_parameters[i1, j1, edge_parameter_index]
-                          + <np.float64_t> alpha[(i0, j0, s0)])
-        alpha[(i0, j0, s0, i1, j1, s1, edge_parameter_index)] = edge_potential
-        alpha[(i1, j1, s1)] = logaddexp(<np.float64_t> alpha.get((i1, j1, s1), -inf),
-                                        edge_potential)
+        edge_potential = (x_dot_parameters[next_state[0], next_state[1], edge_parameter_index]
+                          + <np.float64_t> alpha[current_state])
+        alpha[edge] = edge_potential
+        alpha[next_state] = logaddexp(<np.float64_t> alpha.get(next_state, -inf), 
+                                      edge_potential)
```
Comment (fgregg) [inline pyhacrf/algorithms.pyx:48]: This is just for legibility, taking advantage of the cython's c-tuple

Comment (dirko) [inline pyhacrf/algorithms.pyx:48]: :+1:

File: MANIFEST.in [hunk 2]
```diff
@@ -1,2 +1,3 @@
 include LICENSE
-include README.rst
+include README.st
```
Comment (dirko) [inline MANIFEST.in]: Did the readme change?

**Comment by dirko** [comment]
Thanks that is impressive!

**Pull request closed without merging**